### PR TITLE
Fix translation of saved default player names

### LINKED ISSUE #4906: Main menu, scenario info language is not updated properly
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

When we switch language and then check new scenario info - player's names are not updated until we switch scenario.
English:
image

Switch to Russian:
image
image

Then choose any other scenario:
image

When we switch back to the first scenario:
image

### Save file

None.

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by oleg-derevenetz**
fix #4906

Please note that with this change the names of the players in savefiles will be either empty or non-default (and default names will be properly translated for loaded savefiles - for example, in the scenario info window). Before this change:

Before

After this change:

After

### CODE REVIEW

File: src/fheroes2/system/save_format_version.h [hunk 1]
```diff
@@ -23,12 +23,13 @@
 enum SaveFileFormat
 {
     // TODO: if you're adding a new version you must assign it to CURRENT_FORMAT_VERSION located at the bottom.
+    FORMAT_VERSION_0912_RELEASE = 9702,
```
Comment (ihhub) [inline src/fheroes2/system/save_format_version.h:26]: We added a new version but don't use it in the code. I think we should have some logic related to it.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @oleg-derevenetz , I left few comments here. Could you please take a look?

File: src/fheroes2/system/save_format_version.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/fheroes2/system/save_format_version.h:26]: Hi @ihhub version field size is just 16bit AFAIK. Am I missing something?

Comment (oleg-derevenetz) [reply src/fheroes2/system/save_format_version.h:26]: @ihhub The new version is required due to the fact that new savefiles will be incompatible with old game releases. But there is no need for any fixups in the case of new releases loading old savefiles.

[the remark this answers is not in the document]
Comment (ihhub) [reply src/fheroes2/system/save_format_version.h:26]: Oh, I missed this. Then this is bad. Shall we put at least 9800?

[the remark this answers is not in the document]
Comment (oleg-derevenetz) [reply src/fheroes2/system/save_format_version.h:26]: OK. But it seems that these 16 bits are exhausting too fast :)

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @oleg-derevenetz , I left one suggestion here. Could you please take a look at it?

**Comment by ihhub** [comment]
@oleg-derevenetz , thank you very much for this fix!

**Pull request merged**